openapi.yaml

```diff
@@ -151,6 +172,40 @@ components:
           example:
             - some_scenario
             - another_scenario
+    ScenarioInfo:
+      required:
+        - edoc
+      type: object
+      properties:
+        edoc:
```

Comment:
Just one more question. Why the `edoc` name for the parameter containing scenario description? I think it may be a bit confusing to an end-user who may not be very familiar with Erlang. Do you think parameter name like `desc` or `doc` would be a good replacement?

Reply:
sure, we can also encapsulate parameters inside some dedicated object to avoid confusions:

```json
{
  "doc": "scenario description",
  "parameters": {
    "configuration_parameter1": {
      "module": "module_name",
      "description": "parameter description",
      "default_value": "default value",
      "verification_fn": "MFA or the list of the valid values",
      "update_fn": "MFA or readonly"
    },
    "configuration_parameter2": {
      "module": "module_name",
      "description": "parameter description",
      "default_value": "default value",
      "verification_fn": "MFA or the list of the valid values",
      "update_fn": "MFA or readonly"
    }
  }
}
```

alternatively we can encapsulate it into the array:

```json
{
  "doc": "scenario description",
  "parameters": [
    {
      "name": "configuration_parameter1",
      "module": "module_name",
      "description": "parameter description",
      "default_value": "default value",
      "verification_fn": "MFA or the list of the valid values",
      "update_fn": "MFA or readonly"
    },
    {
      "name": "configuration_parameter2",
      "module": "module_name",
      "description": "parameter description",
      "default_value": "default value",
      "verification_fn": "MFA or the list of the valid values",
      "update_fn": "MFA or readonly"
    }
  ]
}
```

Not sure which way is better.

Reply:
Encapsulating parameters inside the `paramaters` key is a good idea, let's do that. Both structures (map or array) inside the key seems fine to me. I'm slightly in favour of a map, but if you find the array to be more comfortable or flexible, that's also fine by me.

Reply:
I'm also in favour of a map, I just hope we're not biased by erlang 🤷‍♂️